server/instrumentation-backend/src/main/java/sh/calaba/instrumentationbackend/actions/text/WebViewInputScripts.java

```diff
@@ -0,0 +1,191 @@
+            "  if (element.selectionStart !== null && typeof element.selectionStart !== 'undefined') {\n" +
+            "    textLength = element.value.length;\n" +
+            "  } else {\n" +
+            "    textLength = element.innerHTML.length;\n" +
+            "  }\n" +
+            "  if (argBeforeLength < 0) {\n" +
+            "    beforeLength = textLength + argBeforeLength + 1;\n" +
+            "  } else {\n" +
+            "    beforeLength = argBeforeLength;\n" +
+            "  }\n" +
+            "  if (argAfterLength < 0) {\n" +
+            "    afterLength = textLength + argAfterLength + 1;\n" +
+            "  } else {\n" +
+            "    afterLength = argAfterLength;\n" +
+            "  }\n" +
+            "  for (var i = 0; i < beforeLength; i++) {\n" +
+            "    removeCharacter(element, true);\n" +
+            "    simulateKeyEvent(element, true);\n" +
+            "  }\n" +
+            "  for (var j = 0; j < afterLength; j++) {\n" +
+            "    removeCharacter(element, false);\n" +
+            "    simulateKeyEvent(element, false);\n" +
+            "  }\n" +
+            "}\n" +
+            "deleteText(%1$s,%2$s);";
+
+    // This script allows to add selection inside active element
+    static final String SelectScript =
+            "function selectText(argFrom, argTo) {\n" +
+            "  var activeElement = document.activeElement;\n" +
+            "  var tagName = activeElement.tagName.toLowerCase();\n" +
+            "  var from;\n" +
+            "  var to;\n" +
+            "  var textLength;\n" +
+            "  if (activeElement.selectionStart !== null && typeof activeElement.selectionStart !== 'undefined') {\n" +
+            "    textLength = activeElement.value.length;\n" +
+            "  } else {\n" +
+            "    textLength = activeElement.innerHTML.length;\n" +
+            "  }\n" +
+            "  if (argFrom < 0) {\n" +
+            "    from = textLength + argFrom + 1;\n" +
+            "  } else {\n" +
+            "    from = argFrom;\n" +
+            "  }\n" +
+            "  if (argTo < 0) {\n" +
+            "    to = textLength + argTo + 1;\n" +
+            "  } else {\n" +
+            "    to = argTo;\n" +
+            "  }\n" +
+            "  if (activeElement.selectionStart !== null && typeof activeElement.selectionStart !== 'undefined') {\n" +
+            "    activeElement.focus();\n" +
+            "    activeElement.setSelectionRange(from, to);\n" +
+            "  } else if (window.getSelection) {\n" +
+            "    var selection = window.getSelection();\n" +
+            "    var range = document.createRange();\n" +
+            "    range.setStart(activeElement.firstChild, from);\n" +
+            "    range.setEnd(activeElement.firstChild, to);\n" +
+            "    selection.removeAllRanges();\n" +
+            "    selection.addRange(range);\n" +
+            "  }\n" +
+            "}\n" +
+            "selectText(%1$s,%2$s);";
+}
```

Comment:
Well structured! :+1: 

I was looking if you could perhaps share code between the scripts by keeping each function in separate variables and then just `+`ing them together in the final scripts, but it didn't look straight forward. Besides - it's not that much code either.

One ask though - could you document which exact string substitution params each script expects?

Reply:
I see solution here to use JS code and parameters strictly:
Instead of generic `public static Result evalWebViewInputScript(WebView servedView, String script, Object ... scriptParams)` method I can create separate methods inside WebViewInputScripts, like:
`public static String enterTextScript(String text)`
`public static String deleteTextScript(int argBeforeLength, int argAfterLength)`

and common script to execute JS will be like this:
`public static Result evalWebViewInputScript(WebView servedView, String script)`

What do you think? Otherwise I will add comments to existing code.

Reply:
I prefer strict methods for JS, so I will do this. Anyway it is fast to change code to use generic method.
About separate vars for js functions: this make sense, but in current implementation these functions are different inside and their code it is not highly reusable, so let's keep 3 vars for different operations.

Reply:
Like this :+1: Type safety FTW!